src/word.c

```diff
@@ -0,0 +1,67 @@
+#include "dictionary.h"
+#include <stdio.h>
+#include <strings.h>
+#include <string.h>
+#include <assert.h>
+#include <stdlib.h>	
+
+int num_punctuation = 22;
+char* punctuation_array[] = {"+",",",":"," ",".","-","'","&","!","?",":",";","#","~","=","/","$","£","^","\n","_","<",">"};
+
+// checks if punctuation exists within array
+int is_in_array(char* punctuation_array[], char* word) {
+    int i = 0;
+	for ( ; i < num_punctuation ; i++) {
+		if (strcmp(punctuation_array[i], word) == 0) {
+			//printf("comparing %s to %s", word, punctuation_array[i]);
+			return 1;	}
+	}
+	return -1;
+}
+
+// checks if word is valid -- does not have erroneous punctuations within
+int valid_word(char* word, dict_t* dict) {
+	 if (is_in_array(punctuation_array, word) == -1 && *word != '\n') {
+	    return dict_exists(dict, word);
+
+	}
+	else {
+		return EXIT_SUCCESS; //automatically assume word is correct as it represents punctuation
```

Comment:
How do we know that punctuation isn't inside the word, like `jkc83.-a734nf`?

Reply:
@sarikam1

Reply:
@dwahme we don't factor out in-word punctuation like slashes and apostrophes, but assume that any punctuation otherwise represents a token to separate words (i.e the function they grammatically serve). We can't pass punctuation like that into the trie because it's not clear how we would formulate the right combination of strings

hi, i d.on't kn.ow, i can't s.pell, if we consider punctuation to possibly be contained within words do we also pass in "hi, i" or "w, i", etc. It's easier to split the word (i.e not pass in separative punctuation) and give the user the option to delete redundant suggestions.

Reply:
@dwahme how do you want me to handle this?

Reply:
Since our dictionary will be able to check for words containing punctuation but not containing spaces, I think using spaces as a separator would be an acceptable solution. What do you think? 

You have a bit more familiarity with how this process works in the actual spellchecker, so as long as it's documented and understandable, I've seen that it works.

Reply:
ok, so we've decided to only use spaces as delimiters, and for other punctuation to check if punctuation if it is trailing or a prefix to a word. if not, it will be inputted into valid_word.